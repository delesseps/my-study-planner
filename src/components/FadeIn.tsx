//Display fade animation after loading
import * as React from "react";
import styled, { keyframes } from "styled-components";

const FadeIn = ({ children }: { children: React.ReactElement }) => {
  return <Wrapper>{children}</Wrapper>;
};

const fadeIn = keyframes`
   from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
`;

const Wrapper = styled.div`
  animation: ${fadeIn} 0.5s;
`;

export default FadeIn;
