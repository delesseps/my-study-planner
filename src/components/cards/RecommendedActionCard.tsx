import React, { useEffect, useState } from "react";
import styled from "styled-components";
import { Badge, Divider } from "antd";
import IEvaluation from "interfaces/IEvaluation";
import IHomework from "interfaces/IHomework";
import { determineColor } from "utils";
import HomeworkDescriptionModal from "components/modals/HomeworkDescription";
import EvaluationDescriptionModal from "components/modals/EvaluationDescription";
import { useDispatch } from "react-redux";
import { editEvaluation, editHomework } from "store/effects";

interface IRecommendedActionCardProps {
  assignment: IHomework | IEvaluation | any;
}

const RecommendedActionCard: React.FC<IRecommendedActionCardProps> = ({
  assignment
}) => {
  const dispatch = useDispatch();
  const [recommendedAction, setRecommendedAction] = useState("");

  useEffect(() => {
    assignment.evaluationType
      ? setRecommendedAction("Start studying for ")
      : setRecommendedAction("Start Working on ");
  }, [assignment]);

  const handleViewMoreClick = () => {
    assignment.evaluationType
      ? EvaluationDescriptionModal(assignment)
      : HomeworkDescriptionModal(assignment);
  };

  const handleMarkAsDone = () => {
    assignment.done = true;
    dispatch(
      (assignment.evaluationType ? editEvaluation : editHomework)(assignment)
    );
  };

  return (
    <Wrapper>
      <RowWrapper>
        <Badge color={determineColor(assignment.urgency)} />
        <RecomendationTitle>
          {recommendedAction + assignment.subject}
        </RecomendationTitle>
      </RowWrapper>

      <ActionContainer>
        <ViewMore onClick={handleViewMoreClick}>View More</ViewMore>
        <Divider type="vertical" />
        <Done onClick={handleMarkAsDone}>Done</Done>
      </ActionContainer>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;

  padding: 2rem 2rem;

  border-top: 0.5px solid ${props => props.theme.fontColors.textRgba(0.2)};
  border-bottom: 0.5px solid ${props => props.theme.fontColors.textRgba(0.2)};

  width: 100%;
`;

const RowWrapper = styled.div`
  display: flex;
  align-items: center;
`;

const RecomendationTitle = styled.h3`
  margin-bottom: 0;
  font-weight: 400;

  letter-spacing: 1px;
  font-size: 1.7rem;
  color: ${props => props.theme.fontColors.textRgba(0.8)};
`;

const ActionContainer = styled.div`
  display: flex;
  align-items: center;
`;

const ViewMore = styled.span`
  cursor: pointer;
  color: ${props => props.theme.colors.main};
  margin-left: 1.3rem;

  &:hover {
    text-decoration: underline;
  }
`;

const Done = styled.span`
  cursor: pointer;
  color: ${props => props.theme.colors.main};

  &:hover {
    text-decoration: underline;
  }
`;

export default RecommendedActionCard;
