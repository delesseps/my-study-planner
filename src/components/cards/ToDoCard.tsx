import React from "react";
import styled from "styled-components";
import { DeleteOutlined } from "@ant-design/icons";
import { Badge, Checkbox, Popconfirm, Tooltip } from "antd";

import { determineColor } from "utils";
import { useToDo } from "features/toDo/toDo-hooks";
import IToDo from "constants/interfaces/IToDo";

interface IToDoCardProps {
  toDo: IToDo;
  index: number;
}

const ToDoCard: React.FC<IToDoCardProps> = ({ toDo, index }) => {
  const {
    remove: [removeMutate],
    edit: [editMutate],
  } = useToDo();

  const handleDeleteClick = () => {
    removeMutate({ id: toDo._id, index });
  };

  const handleDoneClick = () => {
    toDo.done = true;
    editMutate({ toDo, index });
  };

  return (
    <Wrapper>
      <RowWrapper>
        <Badge color={determineColor(toDo.urgency)} />
        <TaskTitle>{toDo.task}</TaskTitle>
      </RowWrapper>
      <RowWrapper>
        <Action>
          <Tooltip title="Delete" mouseEnterDelay={0.4}>
            <Popconfirm
              title="Are you sure to delete this to-do?"
              arrowPointAtCenter={true}
              okText="Yes"
              cancelText="No"
              onConfirm={handleDeleteClick}
            >
              <DeleteIcon />
            </Popconfirm>
          </Tooltip>
        </Action>
        <Action>
          <Tooltip title="Done" mouseEnterDelay={0.4}>
            <Checkbox onChange={handleDoneClick} />
          </Tooltip>
        </Action>
      </RowWrapper>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;

  padding: 2rem 2rem;

  border-top: 0.5px solid ${(props) => props.theme.fontColors.textRgba(0.2)};
  border-bottom: 0.5px solid ${(props) => props.theme.fontColors.textRgba(0.2)};

  width: 100%;
`;

const RowWrapper = styled.div`
  display: flex;
  align-items: center;
`;

const Action = styled.button`
  border: none;
  background: transparent;

  outline: none;
`;

const TaskTitle = styled.h3`
  text-transform: capitalize;
  margin-bottom: 0;
  font-weight: 400;

  letter-spacing: 1px;
  font-size: 1.7rem;
  color: ${(props) => props.theme.fontColors.textRgba(0.8)};
`;

const DeleteIcon = styled(DeleteOutlined)`
  cursor: pointer;
  font-size: 1.9rem;
  transition: 0.1s;
  margin-right: 1.5rem;

  &:hover {
    color: ${(props) => props.theme.colors.main};
  }
`;

export default ToDoCard;
