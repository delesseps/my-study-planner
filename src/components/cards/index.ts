export { default as EvaluationCard } from "./EvaluationCard";
export { default as HomeworkCard } from "./HomeworkCard";
export { default as RecommendedActionCard } from "./RecommendedActionCard";
export { default as ToDoCard } from "./ToDoCard";
