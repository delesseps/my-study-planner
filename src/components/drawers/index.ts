export { default as EvaluationDrawer } from "./EvaluationDrawer";
export { default as HomeworkDrawer } from "./HomeworkDrawer";
export { default as ToDoDrawer } from "./ToDoDrawer";
export { default as CourseDrawer } from "./CourseDrawer";
