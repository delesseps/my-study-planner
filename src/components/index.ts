export { default as BannerAuth } from "./BannerAuth";
export { default as FadeIn } from "./FadeIn";
export { default as Loading } from "./Loading";
export { default as OAuthButtons } from "./OAuthButtons";
export { default as Section404 } from "./Section404";
export { default as Sidebar } from "./Sidebar";
export { default as SignInForm } from "./SignInForm";
export { default as SignUpForm } from "./SignUpForm";
export { default as TopBar } from "./TopBar";
export { default as UploadPicture } from "./UploadPicture";
export { default as FullPageErrorFallback } from "./FullPageErrorFallback";
export { default as Profiler } from "./Profiler";
