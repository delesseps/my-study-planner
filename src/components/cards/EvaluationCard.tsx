import React, { useState } from "react";
import styled from "styled-components";
import { Icon, Badge, Avatar, Divider, Popconfirm, Tooltip } from "antd";
import IEvaluation from "interfaces/IEvaluation";
import { setDate, determinePriority, determineColor } from "utils";
import moment from "moment";
import { useDispatch } from "react-redux";
import { deleteEvaluation, editEvaluation } from "store/effects";
import EvaluationDescriptionModal from "components/modals/EvaluationDescription";

const EvaluationDrawer = React.lazy(() =>
  import("components/drawers/EvaluationDrawer")
);

interface IEvaluationCardProps {
  evaluation: IEvaluation;
  index: number;
}

const EvaluationCard: React.FunctionComponent<IEvaluationCardProps> = ({
  evaluation,
  index
}) => {
  const [visibleEdit, setVisibleEdit] = useState(false);
  const dispatch = useDispatch();

  const handleViewMoreClick = () => {
    EvaluationDescriptionModal(evaluation);
  };

  const handleEditClick = () => {
    setVisibleEdit(true);
  };

  const handleDeleteClick = () => {
    dispatch(deleteEvaluation(evaluation._id, index));
  };

  const handleDoneClick = () => {
    evaluation.done = true;
    dispatch(editEvaluation(evaluation, index));
  };

  return (
    <Wrapper>
      <EvaluationDrawer
        visibleEdit={visibleEdit}
        setVisibleEdit={setVisibleEdit}
        evaluation={evaluation}
        index={index}
      />
      <MainInfo>
        <Assignment>
          <AssignmentTitle>
            <Capitalize>{evaluation.evaluationType}</Capitalize>:{" "}
            {evaluation.subject}
          </AssignmentTitle>
          <AssignmentPriority>
            <Badge color={determineColor(evaluation.urgency)} />
            {determinePriority(evaluation.urgency)}
          </AssignmentPriority>
        </Assignment>
        <Actions>
          <Tooltip title="Done">
            <StyledIcon onClick={handleDoneClick} type="check" />
          </Tooltip>

          <Tooltip title="Edit">
            <StyledIcon onClick={handleEditClick} type="edit" />
          </Tooltip>

          <Popconfirm
            title="Are you sure to delete this evaluation?"
            arrowPointAtCenter={true}
            placement="topRight"
            okText="Yes"
            cancelText="No"
            onConfirm={handleDeleteClick}
          >
            <StyledIcon type="delete" />
          </Popconfirm>
        </Actions>
      </MainInfo>
      <OtherInfo>
        <User>
          <Avatar size={30} icon="user" src={evaluation.createdBy.picture} />
          <UserName>{evaluation.createdBy.name}</UserName>
        </User>
        <Date>
          <Clock type="clock-circle" />
          {setDate(moment(evaluation.date))} <Divider type="vertical" />
          <ViewMore onClick={handleViewMoreClick}> View More</ViewMore>
        </Date>
      </OtherInfo>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  padding: 0.8rem 2rem;
  border: 0.7px solid ${props => props.theme.fontColors.textRgba(0.15)};
  border-radius: 5px;

  display: flex;
  flex-direction: column;

  height: 16rem;
  width: 100%;

  justify-content: space-between;
`;

const MainInfo = styled.div`
  display: flex;
  justify-content: space-between;
`;

const Assignment = styled.div`
  display: flex;
  flex-direction: column;
`;

const AssignmentTitle = styled.h3`
  letter-spacing: 0.5px;
  font-weight: 500;
  font-size: 1.7rem;
  color: ${props => props.theme.fontColors.textRgba(0.8)};
`;

const Capitalize = styled.span`
  text-transform: capitalize;
`;

const AssignmentPriority = styled.h5`
  display: flex;
  align-items: center;
  color: ${props => props.theme.fontColors.textRgba(0.5)};
  margin: 0;
`;

const Actions = styled.div`
  display: flex;
  align-items: center;
  & > *:not(:last-child) {
    margin-right: 1.5rem;
  }
`;

const StyledIcon = styled(Icon)`
  cursor: pointer;
  font-size: 1.9rem;
  transition: 0.1s;

  &:hover {
    color: ${props => props.theme.colors.main};
  }
`;

const OtherInfo = styled.div`
  display: flex;
  justify-content: space-between;
`;

const User = styled.div`
  display: flex;
  align-items: center;
`;

const UserName = styled.h5`
  letter-spacing: 1px;
  display: flex;
  font-weight: bold;
  font-size: 1.2rem;
  align-items: center;
  color: ${props => props.theme.fontColors.textRgba(0.6)};
  margin: 0;
  margin-left: 0.7rem;
`;

const Date = styled.h5`
  letter-spacing: 1px;
  display: flex;
  font-weight: 400;
  align-items: center;
  color: ${props => props.theme.fontColors.textRgba(0.8)};
  margin: 0;
`;

const Clock = styled(Icon)`
  font-size: 2rem;
  margin-right: 0.7rem;
`;

const ViewMore = styled.span`
  cursor: pointer;
  color: ${props => props.theme.colors.main};
  margin-left: 0.5rem;

  &:hover {
    text-decoration: underline;
  }
`;

export default EvaluationCard;
