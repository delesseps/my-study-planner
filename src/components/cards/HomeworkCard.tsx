import React, { useState } from "react";
import styled from "styled-components";
import {
  UserOutlined,
  CheckOutlined,
  EditOutlined,
  DeleteOutlined,
  ClockCircleOutlined,
} from "@ant-design/icons";
import { Badge, Avatar, Divider, Popconfirm, Tooltip } from "antd";
import moment from "moment";
import { useToggle } from "react-use";
import { useDispatch } from "react-redux";

import { setDate, determinePriority, determineColor } from "utils";
import { deleteHomework, editHomework } from "store/effects";
import IHomework from "constants/interfaces/IHomework";
import HomeworkDescriptionModal from "components/modals/HomeworkDescription";
import { useHomework } from "features/homework/homework-hooks";

const HomeworkDrawer = React.lazy(() =>
  import("components/drawers/HomeworkDrawer")
);

interface IHomeworkCardProps {
  homework: IHomework;
  index: number;
}

const HomeworkCard: React.FunctionComponent<IHomeworkCardProps> = ({
  homework,
  index,
}) => {
  const [openDrawer, toggleDrawer] = useToggle(false);
  const {
    edit: [editMutate],
    remove: [removeMutate],
  } = useHomework();

  const handleViewMoreClick = () => {
    HomeworkDescriptionModal(homework);
  };

  const handleEditClick = () => {
    toggleDrawer(true);
  };

  const handleDeleteClick = () => {
    removeMutate({ id: homework._id, index });
  };

  const handleDoneClick = () => {
    homework.done = true;
    editMutate({ homework, index });
  };

  return (
    <Wrapper>
      <HomeworkDrawer
        visible={openDrawer}
        setVisible={toggleDrawer}
        homework={homework}
        index={index}
      />
      <MainInfo>
        <Assignment>
          <AssignmentTitle>{homework.subject}</AssignmentTitle>
          <AssignmentPriority>
            <Badge color={determineColor(homework.urgency)} />
            {determinePriority(homework.urgency)}
          </AssignmentPriority>
        </Assignment>
        <Actions>
          <Tooltip title="Done">
            <CheckIcon onClick={handleDoneClick} />
          </Tooltip>

          <Tooltip title="Edit">
            <EditIcon onClick={handleEditClick} />
          </Tooltip>

          <Popconfirm
            title="Are you sure to delete this homework?"
            arrowPointAtCenter={true}
            placement="topRight"
            okText="Yes"
            cancelText="No"
            onConfirm={handleDeleteClick}
          >
            <DeleteIcon />
          </Popconfirm>
        </Actions>
      </MainInfo>
      <OtherInfo>
        <User>
          <Avatar
            size={30}
            icon={<UserOutlined />}
            src={homework.createdBy.picture}
          />
          <UserName>{homework.createdBy.name}</UserName>
        </User>
        <Date>
          <ClockIcon />
          {setDate(moment(homework.date))} <Divider type="vertical" />
          <ViewMore onClick={handleViewMoreClick}> View More</ViewMore>
        </Date>
      </OtherInfo>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  padding: 0.8rem 2rem;
  border: 0.7px solid ${(props) => props.theme.fontColors.textRgba(0.15)};
  border-radius: 5px;

  display: flex;
  flex-direction: column;

  height: 16rem;
  width: 100%;

  justify-content: space-between;
`;

const MainInfo = styled.div`
  display: flex;
  justify-content: space-between;
`;

const Assignment = styled.div`
  display: flex;
  flex-direction: column;
`;

const AssignmentTitle = styled.h3`
  letter-spacing: 0.5px;
  font-weight: 500;
  font-size: 1.7rem;
  color: ${(props) => props.theme.fontColors.textRgba(0.8)};
`;

const AssignmentPriority = styled.h5`
  display: flex;
  align-items: center;
  color: ${(props) => props.theme.fontColors.textRgba(0.5)};
  margin: 0;
`;

const Actions = styled.div`
  display: flex;
  align-items: center;
  & > *:not(:last-child) {
    margin-right: 1.5rem;
  }
`;

const CheckIcon = styled(CheckOutlined)`
  cursor: pointer;
  font-size: 1.9rem;
  transition: 0.1s;

  &:hover {
    color: ${(props) => props.theme.colors.main};
  }
`;
const EditIcon = styled(EditOutlined)`
  cursor: pointer;
  font-size: 1.9rem;
  transition: 0.1s;

  &:hover {
    color: ${(props) => props.theme.colors.main};
  }
`;
const DeleteIcon = styled(DeleteOutlined)`
  cursor: pointer;
  font-size: 1.9rem;
  transition: 0.1s;

  &:hover {
    color: ${(props) => props.theme.colors.main};
  }
`;

const OtherInfo = styled.div`
  display: flex;
  justify-content: space-between;
`;

const User = styled.div`
  display: flex;
  align-items: center;
`;

const UserName = styled.h5`
  letter-spacing: 1px;
  display: flex;
  font-weight: bold;
  font-size: 1.2rem;
  align-items: center;
  color: ${(props) => props.theme.fontColors.textRgba(0.6)};
  margin: 0;
  margin-left: 0.7rem;
`;

const Date = styled.h5`
  letter-spacing: 1px;
  display: flex;
  font-weight: 400;
  align-items: center;
  color: ${(props) => props.theme.fontColors.textRgba(0.8)};
  margin: 0;
`;

const ClockIcon = styled(ClockCircleOutlined)`
  font-size: 2rem;
  margin-right: 0.7rem;
`;

const ViewMore = styled.span`
  cursor: pointer;
  color: ${(props) => props.theme.colors.main};
  margin-left: 0.5rem;

  &:hover {
    text-decoration: underline;
  }
`;

export default HomeworkCard;
