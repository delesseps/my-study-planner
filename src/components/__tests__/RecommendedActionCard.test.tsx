import React from "react";
import {
  render,
  fireEvent,
  waitForElement,
  waitForElementToBeRemoved
} from "test-utils";
import { RecommendedActionCard } from "components/cards";

test("renders correctly", () => {
  const props = {
    assignment: {
      _id: "",
      subject: "homework",
      date: "",
      urgency: "normal",
      description: "Extra hard homework",
      done: false,
      createdBy: {
        _id: "",
        name: "",
        picture: ""
      }
    }
  };

  const { asFragment } = render(<RecommendedActionCard {...props} />);
  expect(asFragment()).toMatchSnapshot();
});

test("will display homework modal with description", async () => {
  const props = {
    assignment: {
      _id: "",
      subject: "Really hard homework",
      date: new Date(),
      urgency: "normal",
      description: "Extra hard homework",
      done: false,
      createdBy: {
        _id: "",
        name: "",
        picture: ""
      }
    }
  };

  const { getByText } = render(<RecommendedActionCard {...props} />);

  fireEvent.click(getByText("View More"));

  await waitForElement(() => getByText(props.assignment.subject));
  getByText(`Description: ${props.assignment.description}`);

  fireEvent.click(getByText("OK"));
});

test("will display homework modal with no description", async () => {
  const props = {
    assignment: {
      _id: "",
      subject: "No description homework",
      date: new Date(),
      urgency: "normal",
      description: null,
      done: false,
      createdBy: {
        _id: "",
        name: "",
        picture: ""
      }
    }
  };

  const { getByText } = render(<RecommendedActionCard {...props} />);
  fireEvent.click(getByText("View More"));

  await waitForElement(() => getByText(props.assignment.subject));
  getByText("No Description");

  fireEvent.click(getByText("OK"));
});

test("will display evaluation modal with description", async () => {
  const props = {
    assignment: {
      _id: "",
      subject: "Hamlet",
      evaluationType: "quiz",
      date: "",
      urgency: "normal",
      description: "chapters",
      done: false,
      createdBy: {
        _id: "",
        name: "",
        picture: ""
      }
    }
  };

  const { getByText } = render(<RecommendedActionCard {...props} />);

  fireEvent.click(getByText("View More"));

  await waitForElement(() => getByText(props.assignment.evaluationType));
  getByText(`: ${props.assignment.subject}`);
  getByText(`Description: ${props.assignment.description}`);

  fireEvent.click(getByText("OK"));
});

test("marks as done", async () => {
  const props = {
    assignment: {
      _id: "",
      subject: "Hamlet",
      evaluationType: "quiz",
      date: "",
      urgency: "normal",
      description: "chapters",
      done: false,
      createdBy: {
        _id: "",
        name: "",
        picture: ""
      }
    }
  };

  const { getByText } = render(<RecommendedActionCard {...props} />);

  fireEvent.click(getByText("Done"));

  waitForElementToBeRemoved(() =>
    getByText(`Start studying for ${props.assignment.subject}`)
  );
});
