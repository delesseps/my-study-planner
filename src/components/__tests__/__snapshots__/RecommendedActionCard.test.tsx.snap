// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 2rem 2rem;
  border-top: 0.5px solid rgba(39,39,39,0.2);
  border-bottom: 0.5px solid rgba(39,39,39,0.2);
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  margin-bottom: 0;
  font-weight: 400;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-size: 1.7rem;
  color: rgba(39,39,39,0.8);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  cursor: pointer;
  color: #00ADB5;
  margin-left: 1.3rem;
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  cursor: pointer;
  color: #00ADB5;
}

.c5:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
      >
        <span
          class="ant-badge-status-dot ant-badge-status-yellow"
        />
        <span
          class="ant-badge-status-text"
        />
      </span>
      <h3
        class="c2"
      >
        Start Working on homework
      </h3>
    </div>
    <div
      class="c3"
    >
      <span
        class="c4"
      >
        View More
      </span>
      <div
        class="ant-divider ant-divider-vertical"
        role="separator"
      />
      <span
        class="c5"
      >
        Done
      </span>
    </div>
  </div>
</DocumentFragment>
`;
